framework/src/parser/Parser.C

```diff
@@ -1814,13 +1814,22 @@ Parser::setVectorParameter<VariableName, VariableName>(
     std::istringstream ss(var_name);
 
     // If we are able to convert this value into a Real, then set a default coupled value
+    // NOTE: parameter must be either all default or no defaults
+    unsigned int mode = 0;
     if (ss >> real_value && ss.eof())
-      /* FIXME: the real_value is assigned to defaultCoupledValue overriding the value assigned
-       * before. Currently there is no functionality to separately assign the correct
-       * "real_value[i]" in InputParameters.*/
-      _current_params->defaultCoupledValue(short_name, real_value);
+    {
+      if (mode == 2)
+        mooseError(
+            "Variable name ", short_name, " contains both default and actual variable names.");
+      mode = 1;
+      _current_params->defaultCoupledValue(short_name, real_value, i);
+    }
     else
     {
+      if (mode == 1)
+        mooseError(
+            "Variable name ", short_name, " contains both default and actual variable names.");
```

Comment:
It would be nice if both error messages could provide more info, so that users can actually find where the problem is in the input file and fix it.  At least a line number would be nice. Or a name of the offending object.  But, I have no idea if this is possible. @permcody or @rwcarlsen ?

Reply:
Follow this pattern in Parser::setScalarParameter:

```c++
        _errmsg += errormsg(_input_filename,
                            _root->find(full_name),
                            "invalid ",
                            format_type,
                            " syntax for parameter: ",
                            full_name,
                            "=",
                            strval) +
                   "\n";
```

Try not to call mooseError or throw here - the `_errmsg` member is used to collect multiple errors and emit them all at once so the user doesn't have to fix one error, run again, fail, fix second error, run again, fail, etc.